packages/core/src/middleware/koa-consent-guard.ts

```diff
@@ -0,0 +1,71 @@
+import { experience } from '@logto/schemas';
+import { type MiddlewareType } from 'koa';
+import { type IRouterParamContext } from 'koa-router';
+import { errors, type Provider } from 'oidc-provider';
+
+import RequestError from '../errors/RequestError/index.js';
+import type Libraries from '../tenants/Libraries.js';
+import type Queries from '../tenants/Queries.js';
+import assertThat from '../utils/assert-that.js';
+
+/**
+ * Guard before allowing auto-consent
```

Comment:
Need some comments here, this middleware should guard one-time-token sign-in only.

Reply:
Comments added